### ISSUE TITLE
Incorrect error parsing

### ISSUE THREAD

**Issue by Cyberax** [aws/aws-sdk-go-v2#1207]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [X] I've gone though the [API reference](https://docs.aws.amazon.com/sdk-for-go/v2/api/)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-go) for answers
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-go-v2/issues) and didn't find any solution
  
**Describe the bug**
Multiple errors are parsed incorrectly, making it impossible to use the new error infrastructure. For example, IAM has this definition:
```
type EntityAlreadyExistsException struct {
	Message *string
}
func (e *EntityAlreadyExistsException) ErrorCode() string             { return "EntityAlreadyExistsException" }
```

While the code returned by the API is actually `EntityAlreadyExists` (no word "Exception").

It looks like I can't trust SDK anymore and I have to return back to pre-Smithy version.

**Comment by nguyengg** [comment]
I was wondering the same thing, I have this code that relies on IAM's CreateRole returning EntityAlreadyExistsException.

```golang
if _, err = h.iamSvc.CreateRole(ctx, &iam.CreateRoleInput{
	AssumeRolePolicyDocument: aws.String(assumeRolePolicyDocument),
	Path:                     aws.String(path),
	RoleName:                 aws.String(roleName),
}); err != nil {
	var e *iamtypes.EntityAlreadyExistsException
	if errors.As(err, &e) {
		log.Printf("WARN role %s already exists\n", roleArn)
		return nil
	}
	return fmt.Errorf("iam CreateRole: %w", err)
}
```

And it just doesn't work. I had to use a workaround by calling IAM's GetRole first but that's unreliable in a distributed system.

**Comment by veezhang** [comment]
The `errorCode` in deserializer‘s `switch` is `EntityAlreadyExistsException`.
In fact, what we get is `EntityAlreadyExists`, so return the `smithy.GenericAPIError`.

https://github.com/aws/aws-sdk-go-v2/blob/main/service/iam/deserializers.go#L1602-L1653

**Comment by Cyberax** [comment]
Ping? This is a critical issue. It's not possible to write reliable error handling using the SDK's own methods.

**Comment by skotambkar** [comment]
Thanks for the ping. This issue has been fixed with release v1.6.0 of service/iam client module. Please update your dependency to v1.6.0 in your go.mod for the fix. 

Please let us know if you still face an issue or have any questions.
